packages/node/src/indexer/benchmark.service.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020-2021 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { OnEvent } from '@nestjs/event-emitter';
+import * as moment from 'moment';
+import 'moment-duration-format';
+import { MetricPayload } from '../prometheus/types';
+import { getLogger } from '../utils/logger';
+import { delay } from '../utils/promise';
+import { BlockPayload } from './types';
+
+const SAMPLING_TIME_VARIANCE = 15;
+const logger = getLogger('benchmark');
+
+export class BenchmarkService {
+  private currentProcessingHeight: number;
+  private currentProcessingTimestamp: number;
+  private targetHeight: number;
+  private lastRegisteredHeight: number;
+  private lastRegisteredTimestamp: number;
+  private blockPerSecond: number;
+  private needReset = true;
+
+  async init(): Promise<void> {
+    if (!this.currentProcessingHeight) {
+      await delay(5);
+    }
+    setInterval(() => this.benchmark(), SAMPLING_TIME_VARIANCE * 1000);
+  }
+
+  private benchmark() {
+    if (this.needReset) {
+      this.resetBlock();
+    } else {
+      logger.debug(
+        `current #${this.currentProcessingHeight}, last #${
+          this.lastRegisteredHeight
+        }, take ${
+          (this.currentProcessingTimestamp - this.lastRegisteredTimestamp) /
```

Comment:
this calculation is done more than once, so pls assign the result to a variable

Reply:
actually this debug log doesn't make sense to me, if we want to know whether processing height is changing, we should add log in indexManager not here